app/views.py

```diff
@@ -19,19 +19,45 @@ class IndexView(TemplateView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        if "type" not in self.request.GET or "q" not in self.request.GET:
+
+        context["type"] = ["challenge", "project"]
+
+        context["tags"] = Tag.objects.all()
+        context["selected_tags"] = []
+
+        if "type" not in self.request.GET or (
+            "q" not in self.request.GET and "tag" not in self.request.GET
+        ):
             context["challenges"] = Challenge.objects.all()[:3]
             context["projects"] = Project.objects.all()[:9]
-            return context
-        query = self.request.GET["q"]
-        if self.request.GET["type"] == "challenge":
-            context["challenges"] = Challenge.objects.all().filter(
-                Q(name__contains=query) | Q(description__contains=query)
-            )
-        elif self.request.GET["type"] == "project":
-            context["projects"] = Project.objects.all().filter(
-                Q(name__contains=query) | Q(description__contains=query)
-            )
+        else:
+            initiative = self.request.GET["type"]
+            context["type"] = [initiative]
+
+            if initiative == "challenge":
+                queryset = Challenge.objects.all()
+
+            elif initiative == "project":
+                queryset = Project.objects.all()
+
+            if "tag" in self.request.GET:
+                query = self.request.GET.getlist("tag")
+
+                for tag in query:
+                    currtags = Tag.objects.filter(name__iexact=tag)
+                    queryset = queryset.filter(Q(tags__in=currtags))
+                    context["selected_tags"] += currtags
+
+            if "q" in self.request.GET:
+                query = self.request.GET["q"]
+                context["q"] = query
+
+                queryset = queryset.filter(
+                    Q(name__icontains=query) | Q(description__icontains=query)
+                )
+
+            context[initiative + "s"] = queryset.distinct()
```

Comment:
This is alongside HTML changes, the template for projects and challenges seems extremely similar. Can we not just call this `objects` and `{% for object in objects %}`.

Reply:
Do you refer to these lines? (If so, that works, will implement)

https://github.com/KuriusMTL/stratus/pull/27/commits/6335f38f581698274814e70c20d8e85e2145e9b3#diff-c80b5e347b555901a695e72476d025b0R50-R110

Reply:
(https://github.com/KuriusMTL/stratus/pull/27/commits/9ed00ea804203bffaf6a1bb9d52c3a0f9ac74a6f)